core/src/main/java/org/mule/time/TimeSupplier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.time;
+
+import java.util.function.Supplier;
+
+/**
+ * A {@link Supplier} which provides the current system time
+ * in milliseconds. This is a singleton class, use {@link #INSTANCE}
```

Comment:
This class simply returns System.currentTimeInMillis(). It's completely stateless and all instances will behave the same. Having multiple instances of it doesn't make much sense memory wise.